packages/vite-plugin-monkey/src/node/option.ts

```diff
@@ -290,6 +290,7 @@ export const resolvedOption = (
     },
     server: {
       mountGmApi: server.mountGmApi ?? false,
+      ssl: server.ssl ?? false,
```

Comment:
vite-plugin-monkey already supports SSL, so no additional SSL configuration is needed. You can verify this by running `pnpm -F example serve`

![image](https://github.com/user-attachments/assets/1cfdcbea-a5d6-48d6-902c-ab3963100deb)

Reply:
我试了下，`https`的这种似乎有点问题？
![image](https://github.com/user-attachments/assets/7f6844d0-90a9-446d-8aca-a0dc988ec9f4)

我觉得`entrySrc`的`protocol`可以不需要根据访问的origin参数来判断

![Clip_2024-07-21_10-51-51](https://github.com/user-attachments/assets/d135fa44-6098-47f9-9185-02eed46fd911)

Reply:
![image](https://github.com/user-attachments/assets/0812d870-9ee1-48f3-82b6-351ee4a1bcf6)

Reply:
vite-plugin-monkey must use the origin parameter to get the correct entry point; otherwise, it will not work properly in the example at https://codesandbox.io/p/devbox/vite-monkey-test-9e0vbi

Reply:
我修改了配置项`ssl`=>`entryUrlProtocol`，默认值`origin`，这样的话就不影响原来的

Reply:
So, what issue does the newly added parameter `entryUrlProtocol` resolve?

Reply:
`entryUrlProtocol `是上面`ssl`的代替配置，它的使用场景是这样的
默认不使用`https`，有些插件(Safari的stay插件)就像我上面的，直接访问`https`链接无法安装脚本，那么
我需要
1. 先启动`http`的来安装脚本
2. 手动修改`entrySrc`的`http`为`https`
3. 然后再启用`https`来运行脚本

有这个配置相当于省略第2个步骤，不需要对server:xxx脚本内容进行修改

Reply:
vite-plugin-monkey already supports SSL

https://github.com/lisonge/vite-plugin-monkey/pull/174#discussion_r1685609562

---

> 我试了下，https 的这种似乎有点问题？

Does your device trust this certificate: https://github.com/lisonge/vite-plugin-monkey/blob/main/playground/example/127.0.0.1.pem ?

Reply:
奇怪，安装了证书但是不信任
不过我自己生成了证书就ok了
![Clip_2024-07-21_12-50-16](https://github.com/user-attachments/assets/a62811c8-cd75-45c0-93ae-c8358cf69ce7)